Source/SwiftLintBuiltInRules/Rules/Lint/UnneededSynthesizedInitializerRule.swift

```diff
@@ -0,0 +1,312 @@
+            }
+            return super.visit(node)
+        }
+
+        override func visit(_ node: InitializerDeclSyntax) -> DeclSyntax {
+            if unneededInitializers.contains(node) {
+                correctionPositions.append(node.positionAfterSkippingLeadingTrivia)
+                let expr: DeclSyntax = ""
+                return expr
+                    .with(\.leadingTrivia, node.leadingTrivia)
+                    .with(\.trailingTrivia, node.trailingTrivia)
+            }
+            return super.visit(node)
+        }
+    }
+}
+
+private extension StructDeclSyntax {
+    var unneededInitializers: [InitializerDeclSyntax] {
+        let unneededInitializers = findUnneededInitializers()
+        let initializersCount = memberBlock.members.filter { $0.decl.is(InitializerDeclSyntax.self) }.count
+        if unneededInitializers.count == initializersCount {
+            return unneededInitializers
+        }
+        return []
+    }
+
+    // Collects all of the initializers that could be replaced by the synthesized
+    // memberwise or default initializer(s).
+    private func findUnneededInitializers() -> [InitializerDeclSyntax] {
+        // swiftlint:disable:previous cyclomatic_complexity
+        var storedProperties: [VariableDeclSyntax] = []
+        var initializers: [InitializerDeclSyntax] = []
+
+        for memberItem in memberBlock.members {
+            let member = memberItem.decl
+            // Collect all stored variables into a list
+            if let varDecl = member.as(VariableDeclSyntax.self) {
+                let modifiers = varDecl.modifiers
+                if modifiers == nil {
+                    storedProperties.append(varDecl)
+                    continue
+                }
+                guard !modifiers.isStatic else { continue }
+                storedProperties.append(varDecl)
+                // Collect any possible redundant initializers into a list
+            } else if let initDecl = member.as(InitializerDeclSyntax.self) {
+                guard initDecl.optionalMark == nil else { continue }
+                guard initDecl.hasThrowsOrRethrowsKeyword == false else { continue }
+                initializers.append(initDecl)
+            }
+        }
+
+        var unneededInitializers = [InitializerDeclSyntax]()
+        for initializer in initializers {
+            guard
+                self.initializerParameters(initializer.parameterList, match: storedProperties)
+            else {
+                continue
+            }
+            guard
+                initializer.parameterList.isEmpty ||
+                initializerBody(initializer.body, matches: storedProperties)
+            else {
+                continue
+            }
+            guard initializerModifiers(initializer.modifiers, match: storedProperties) else {
+                continue
+            }
+            guard initializer.isInlinable == false else {
+                continue
+            }
```

Comment:
```suggestion
            if initializer.isInlinable {
                continue
            }
```
or try to merge all `guard` statements into one.